From a67fcd71a38d9662b5cb2dc4054de19768df2271 Mon Sep 17 00:00:00 2001
From: Weixin Zhou <zwx@rock-chips.com>
Date: Sun, 7 Feb 2021 11:04:37 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3566-rk817-eink-w6: adjust configs
 for lower consumption

1. set vcc3v3_pmu suspend voltage 3v
2. add touch regulator

Signed-off-by: Weixin Zhou <zwx@rock-chips.com>
Change-Id: Ic5ac610bea36675ad8f9dbe511a9ab604b30aa92
---
 .../dts/rockchip/rk3566-rk817-eink-w6.dts     | 20 +++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-rk817-eink-w6.dts b/arch/arm64/boot/dts/rockchip/rk3566-rk817-eink-w6.dts
index 1e1b9d87adc1..fdd73f8f727c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-rk817-eink-w6.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-rk817-eink-w6.dts
@@ -99,6 +99,15 @@
 		regulator-boot-on;
 	};
 
+	vcc_tp: vcc-tp-regulator {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpio = <&gpio0 RK_PC0 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&vcc_tp_en>;
+		regulator-name = "vcc_tp";
+	};
+
 	dummy_codec: dummy-codec {
 		compatible = "rockchip,dummy-codec";
 		#sound-dai-cells = <0>;
@@ -454,7 +463,7 @@
 				regulator-name = "vcc3v3_pmu";
 				regulator-state-mem {
 					regulator-on-in-suspend;
-					regulator-suspend-microvolt = <3300000>;
+					regulator-suspend-microvolt = <3000000>;
 					regulator-changeable-in-suspend;
 				};
 			};
@@ -629,7 +638,7 @@
 		max-x = <1024>;
 		max-y = <758>;
 		tp-size = <9111>;
-		tp-supply = <&vcc3v3_pmu>;
+		tp-supply = <&vcc_tp>;
 		wakeup-source;
 		touchscreen-key-map = <158>; //KEY_HOMEPAGE=172,KEY_BACK=158,KEY_MENU=139
 		goodix,driver-send-cfg = <0>;
@@ -652,6 +661,7 @@
 		reg = <0x38>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&tp_gpio>;
+		vdd-supply = <&vcc_tp>;
 		focaltech,reset-gpio = <&gpio0 RK_PA5 GPIO_ACTIVE_HIGH>;
 		focaltech,irq-gpio = <&gpio0 RK_PA6 IRQ_TYPE_EDGE_FALLING>;
 		focaltech,max-touch-number = <5>;
@@ -768,6 +778,12 @@
 		};
 	};
 
+	vcc-tp {
+		vcc_tp_en: vcc-tp-en {
+			rockchip,pins = <0 RK_PC0 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
 	wireless-wlan {
 		wifi_vbat: wifi-vbat {
 			rockchip,pins = <0 RK_PA0 RK_FUNC_GPIO &pcfg_pull_none>;
-- 
2.35.3

